lib/web/apiserver.go

```diff
@@ -1014,6 +1014,7 @@ func (h *Handler) bindDefaultEndpoints() {
 	h.GET("/webapi/sites/:site/integrations/:name/discoveryrules", h.WithClusterAuth(h.integrationDiscoveryRules))
 	h.GET("/webapi/sites/:site/integrations/:name/ca", h.WithClusterAuth(h.integrationsExportCA))
 	h.DELETE("/webapi/sites/:site/integrations/:name_or_subkind", h.WithClusterAuth(h.integrationsDelete))
+	h.DELETE("/webapi/sites/:site/integration/withcleanup", h.WithClusterAuth(h.integrationsDeleteWithCleanup))
```

Comment:
Why is a separate endpoint necessary? Are there legitimate cases where a user would want to delete an integration but leave the resources it created behind?

Reply:
you're right, i would still need a new endpoint (or create a v2 endpoint of the same endpoint) to handle request succeeding in older proxies

Reply:
I think we can assume Auths are upgraded before Proxies. Then if the newer WebUI hits older proxy, with the new Auth change, the API will return `git servers associated with integration %s must be deleted first`. Do you think if that's a good enough error so we can avoid introducing a new version of the endpoint?

Another point of view is if some of the proxies _are old_, the new web enrolment flow would probably have failed anyway. If they added integration using `tctl` (following the official guide) and see them in the WebUI, then they should know how to do the cleanup using `tctl`.

Reply:
> Do you think if that's a good enough error so we can avoid introducing a new version of the endpoint?

hrm, it'll be an inaccurate error though, because the error is because the old proxy doesn't process the new query param

> the new web enrollment flow would probably have failed anyway

it could succeed, if the new proxy was the one that served the web UI and the requests goes to the new proxy